### PR TITLE
Handle the result of a metrics ping with a log message

### LINKED ISSUE #145: "Uncaught (in promise) Metrics not sent due to DNT" error in Chrome
Opened a theme URL locally in Chrome (64?), changed the "Theme Texture", and got the following in my DevTools console:

```
Uncaught (in promise) Metrics not sent due to DNT.                  index.js:175
```

themer-chrome

firefox_themer_chromer

Digging deeper, this may be coming from the [**testpilot-ga**](https://github.com/mozilla/testpilot-ga/blob/8f53d89b18d1d4745b3cff4251afd924802a596d/src/TestPilotGA.js#L101-L108) source... But not sure if we should be trying to catch upstream errors in our source or not.

---

**UPDATE:** Looks like the easiest/cheapest fix is to add a `.catch()` in: https://github.com/mozilla/Themer/blob/cf279fac5a7f1a81a937884a8569c61215585c42/src/lib/metrics.js#L77-L81

Something like:

```js
  sendEvent(...params) {
    if (analytics) {
      analytics.sendEvent(...params).catch(err => console.dir(err));
    }
  },
```

I'm not sure what's happening w/ the `err` response from [**testpilot-ga**'s `analytics.sendEvent()`](https://github.com/mozilla/testpilot-ga/blob/8f53d89b18d1d4745b3cff4251afd924802a596d/src/TestPilotGA.js#L107). It doesn't look/smell like an Error object, and feels more like a string...

### PR DESCRIPTION
**Pull request by lmorchard**
Fixes #145

### CODE REVIEW

**Comment by lmorchard** [comment]
This still results in an error message in the console when DNT is active. But, it should be slightly more informative like this:
```
[FirefoxColor metrics] sendEvent (ERROR) undefined 
Array(3) [ "receive-theme", "button-click", {…} ]
 Metrics not sent due to DNT.
```

File: src/lib/metrics.js [hunk 1]
```diff
@@ -75,9 +75,15 @@ const Metrics = {
...
   sendEvent(...params) {
-    if (analytics) {
-      analytics.sendEvent(...params);
+    if (!analytics) {
+      return;
     }
+    analytics
+      .sendEvent(...params)
+      .then(response => log("sendEvent (success)", params, response))
+      .catch((response, err) =>
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply src/lib/metrics.js]: true https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

[the remark this answers is not in the document]
Comment (lmorchard) [reply src/lib/metrics.js]: Oh, I was going by [the usage docs for testpilot-ga](https://github.com/mozilla/testpilot-ga/tree/c4119b6a98089c9f0c53b03feb9307bd3f1220a7#usage):
```javascript
analytics
  .sendEvent('ec_value', 'ea_value', {
    cd1: 'cd1_value',
    cm1: 'cm1_value'
  })
  .then(response => {
    console.log('Event successfully sent', response)
  })
  .catch((response, err) => {
    console.error('Event failed while sending', response, err)
  });
```
Maybe those need a tweak?

**Comment by lmorchard** [comment]
Updated to accept only the `err` parameter in `.catch()`

File: src/lib/metrics.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdehaan) [reply src/lib/metrics.js]: Submitted upstream https://github.com/mozilla/testpilot-ga/issues/16 for the devilishly handsome @chuckharmston to consider and educate me.

**Comment by chuckharmston** [comment]
Bouncing this over to @meandavejustice, who is taking over `testpilot-ga` for me.

File: src/lib/metrics.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chuckharmston) [reply src/lib/metrics.js]: Bouncing this over to @meandavejustice, who is taking over `testpilot-ga` for me.

**Comment by johngruen** [comment]
looks good to me...
everything is erroring, but i think that's b/c i'm testing locally
screen shot 2018-04-20 at 10 07 04 am

**Pull request merged**